sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -550,13 +550,19 @@ private HttpPipeline setupPipeline(HttpClient httpClient) {
      */
     public Mono<MsalToken> authenticateWithAuthorizationCode(TokenRequestContext request, String authorizationCode,
                                                              URI redirectUrl) {
-        return publicClientApplicationAccessor.getValue()
-                .flatMap(pc -> Mono.fromFuture(() -> pc.acquireToken(
-                AuthorizationCodeParameters.builder(authorizationCode, redirectUrl)
-                    .scopes(new HashSet<>(request.getScopes()))
-                    .build()))
-                .onErrorMap(t -> new ClientAuthenticationException("Failed to acquire token with authorization code",
-                    null, t)).map(ar -> new MsalToken(ar, options)));
+        AuthorizationCodeParameters parameters = AuthorizationCodeParameters.builder(authorizationCode, redirectUrl)
+            .scopes(new HashSet<>(request.getScopes()))
+            .build();
+        Mono<IAuthenticationResult> acquireToken;
+        if (clientSecret != null) {
```

Comment:
Where are you setting this client secret  in ConfidentialClient ?

Reply:
per offline discussion, this is set in the constructor of the IdentityClient.